### PR TITLE
B get discogs vinyls list by username

### PR DESCRIPTION
**Pull request by AlexKievDeveloper**

### CODE REVIEW

File: src/main/java/com/vinylteam/vinyl/Starter.java [hunk 1]
```diff
@@ -1,62 +1,94 @@
...
-import org.eclipse.jetty.util.resource.PathResource;
+import org.eclipse.jetty.util.resource.JarFileResource;
+import org.eclipse.jetty.util.resource.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.List;
 import java.util.Map;
 
-import static java.lang.Integer.parseInt;
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/Starter.java:28]: You still use parseInt, but I can't see the import

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class DefaultDiscogsService implements DiscogsService {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final ObjectMapper objectMapper;
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java:19]: Reminder that we agreed on an empty row after classes' '{' and before closing '}'.

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        return List.of();
+    }
+
+    Optional<List<DiscogsVinylInfo>> parseDiscogsWantList(String discogsWantList) {
+        try {
+            RawResponse rawResponse = objectMapper.readValue(discogsWantList, RawResponse.class);
+            return rawResponse.getVinylsInfo();
+        } catch (JsonProcessingException e) {
+            logger.error("Want list: {}", discogsWantList);
+            throw new RuntimeException("Exception while want list json processing", e);
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java]: Format is {'wantList':{}}

File: src/test/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsServiceITest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        assertEquals("No Freedom No Liberty", actualOptionalWantList.get().get(1).getRelease());
+        assertEquals("The Detectors", actualOptionalWantList.get().get(1).getArtist());
+        assertEquals("Жаль, Нет Ружья", actualOptionalWantList.get().get(2).getRelease());
+        assertEquals("Король И Шут", actualOptionalWantList.get().get(2).getArtist());
+    }
+
+    @Test
+    @DisplayName("Returns list of releases names from want list")
+    void getVinylsByReleasesFromDB() {
+        //prepare
```
Comment (Greengrer) [inline src/test/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsServiceITest.java]: Maybe just getVinylsByReleasesTest()?

**Review by TarasOrlovskyi — DISMISSED** [review]

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    Optional<List<DiscogsVinylInfo>> parseDiscogsWantList(String discogsWantList) {
+        try {
+            if (discogsWantList != null) {
+                RawResponse rawResponse = objectMapper.readValue(discogsWantList, RawResponse.class);
+                return rawResponse.getVinylsInfo();
+            }
+            return Optional.empty();
+        } catch (JsonProcessingException e) {
+            logger.error("{'WantList':{}}", discogsWantList);
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java]: Add error to logger to have stacktrace

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        }
+    }
+
+    List<String> getVinylsReleases(List<DiscogsVinylInfo> discogsVinylInfoList) {
+        List<String> vinylsReleasesList = new ArrayList<>();
+
+        for (DiscogsVinylInfo discogsVinylInfo : discogsVinylInfoList) {
+            vinylsReleasesList.add(discogsVinylInfo.getRelease());
+        }
+        return vinylsReleasesList;
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java:76]: Add logger.debug("Resulting vinyls releases list is {'releasesList':{}}", vinylsReleasesList);
before return.

File: src/main/java/com/vinylteam/vinyl/util/PropertiesReader.java [hunk 7]
```diff
@@ -16,10 +16,9 @@
     private final String independentPropertiesFile = "application.properties";
 
     public PropertiesReader() {
-        String beginningOfErrorMessage = "Error during loading properties from";
-
+        String beginningOfErrorMessage = "Error during loading properties from ";
         try (InputStream inputStream = getClass().getClassLoader()
-                .getResourceAsStream(independentPropertiesFile)) {
+                .getResourceAsStream("application.properties")) {
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/util/PropertiesReader.java]: If you don't use indpendentPropertiesFile, delete it.(row 16).

File: src/main/java/com/vinylteam/vinyl/util/PropertiesReader.java [hunk 8]
```diff
@@ -16,10 +16,9 @@
...
             logger.debug("Created input stream from file {'inputStream':{}, 'fileName':{}}",
                     inputStream, independentPropertiesFile);
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/util/PropertiesReader.java]: Decide if you use independentPropertiesFile string or just use "application.properties" everywhere

File: src/main/java/com/vinylteam/vinyl/util/PropertiesReader.java [hunk 9]
```diff
@@ -35,11 +34,11 @@ public PropertiesReader() {
         String env = System.getenv("env");
         logger.debug("Created and initialised String env with the value " +
                 "of environmental variable \"env\" {'env':{}}", env);
-        if (env == null) {
+        if (System.getenv("env") == null) {
             dependentOnEnvPropertiesFile = "dev.application.properties";
 
             try (InputStream inputStream = getClass().getClassLoader()
-                    .getResourceAsStream(dependentOnEnvPropertiesFile)) {
+                    .getResourceAsStream("dev.application.properties")) {
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/util/PropertiesReader.java]: I think you are using the version that accidentally changed PropertiesReader to how it was before logger.

Do you use String env?

File: src/main/java/com/vinylteam/vinyl/util/PropertiesReader.java [hunk 10]
```diff
@@ -92,37 +91,15 @@ public PropertiesReader() {
...
     private void validateInputStream(InputStream inputStream) {
         if (inputStream == null) {
             RuntimeException e = new RuntimeException();
-            logger.error(".properties file not found, input stream is null", e);
+            logger.error(".properties file not found.", e);
             throw e;
         }
-        logger.debug("Input stream is valid {'inputStream':{}}", inputStream);
     }
-
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/util/PropertiesReader.java]: Why delete debug that tells that input stream is valid?

Add empty row before end of class.

**Review by Greengrer — CHANGES_REQUESTED** [review]

**Review by TarasOrlovskyi — DISMISSED** [review]

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+        this.USER_AGENT = userAgent;
+        this.CALLBACK_URL = callbackUrl;
+        this.discogsClient = new DiscogsClient(CONSUMER_KEY, CONSUMER_SECRET, USER_AGENT, CALLBACK_URL);
+        this.objectMapper = objectMapper;
+
+        try {
+            discogsClient.getRequestToken();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            logger.error("Failed to connect to discogs user with {'consumer key': {}}, {'consumer secret': {}}, {'user agent': {}}," +
+                    " {'callback url': {}} ", CONSUMER_KEY, CONSUMER_SECRET, USER_AGENT, CALLBACK_URL, e);
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java]: {'consumerKey': {}, 'consumerSecret': {}, 'userAgent': {}, 'callbackUrl': {}}
Sorry I was unclear.

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+    Optional<List<DiscogsVinylInfo>> parseDiscogsWantList(String discogsWantList) {
+        try {
+            if (discogsWantList != null) {
+                RawResponse rawResponse = objectMapper.readValue(discogsWantList, RawResponse.class);
+                return rawResponse.getVinylsInfo();
+            }
+            return Optional.empty();
+        } catch (JsonProcessingException e) {
+            logger.error("{'WantList':{}}", discogsWantList, e);
+            throw new RuntimeException("Exception while want list json processing", e);
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java]: Starting with lowercase, like variable. "{'wantList':{}}"

**Review by Greengrer — CHANGES_REQUESTED** [review]

**Review by TarasOrlovskyi — DISMISSED** [review]

**Review by Greengrer — APPROVED** [review]

**Review by TarasOrlovskyi — APPROVED** [review]

**Pull request merged**